src/components/PhotosGallery/PhotosGallery.jsx

```diff
@@ -0,0 +1,124 @@
+import Image from "next/image"
+import React, { useEffect, useState } from "react"
+
+function PhotosGallery(props) {
+  const [file, setFile] = useState(null)
+
+  const onChangeImageHandler = (e) => {
+    if (e.target.files[0]) {
+      const reader = new FileReader()
+      reader.onload = () => {
+        setFile(reader.result)
+      }
+      reader.readAsDataURL(e.target.files[0])
```

Comment:
It would be better to use the optional chaining operator ("?." or ".") when accessing the file, so that you can avoid errors when the file is null or undefined:
`if (e.target.files?.[0])`